boot/zephyr/Kconfig

```diff
@@ -60,14 +60,6 @@ config BOOT_SIGNATURE_TYPE_ECDSA_P256
 
 endchoice
 
-config BOOT_SIGNATURE_KEY_FILE
```

Comment:
Why this property can't be there?
Removal will impose possible problems with any future upmerge (it is error prone).
I think dependency enlargement to  `MCUBOOT || BOOTLOADER_MCUBOOT` can go upstream.

Reply:
It needs to visible for both the application including MCUboot, and MCUboot itself.
This because the application will use this configuration to invoke the `imgtool.py` script with the correct key file.

When there is support for sharing specific Kconfig options between images it would be possible to have this config here.

Could you explain why moving this option can become an issue?

Reply:
I was not aware of counterpart PR to zephyr upstream at the moment I had ben asking.
Will be still possible to build as stan-alone project. I mean to build only mcuboot as we use to do so fare.

Reply:
My bad, the details about the counterpart PR should have been stated more clearly.

The "normal" way of building MCUboot is not affected by this.